### PR TITLE
Migrate instance admin API to StatusOr<>.

### LINKED ISSUE #1967: Migrate InstanceAdmin to StatusOr<>

### PR DESCRIPTION
**Pull request by dopiera**
This fixes #1967.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/1992)

### CODE REVIEW

File: google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc [hunk 1]
```diff
@@ -63,20 +63,23 @@ void RunInstanceOperations(std::string project_id, int argc, char* argv[]) {
...
             return i.name() == instance_name;
           });
-  if (!instances.failed_locations.empty()) {
+  if (!instances->failed_locations.empty()) {
     std::cerr
         << "The service tells us it has no information about these locations:";
-    for (std::string const& location : instances.failed_locations) {
-      std::cout << " " << location << "\n";
+    for (std::string const& location : instances->failed_locations) {
+      std::cout << " " << location;
```
Comment (devjgm) [inline google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc]: The leading sentence before the colon is sent to `std::cerr`, but then each location here is sent to `std::cout`. Is that intentional? Perhaps this one in the loop should also be `std::cerr`?

Note: "continuing anyway" is also sent to `cerr`.

File: google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc [hunk 2]
```diff
@@ -107,13 +110,16 @@ void RunInstanceOperations(std::string project_id, int argc, char* argv[]) {
...
+  for (auto const& instance : instances_after->instances) {
     std::cout << instance.name() << "\n";
   }
-  if (!instances_after.failed_locations.empty()) {
+  if (!instances_after->failed_locations.empty()) {
     std::cerr
         << "The service tells us it has no information about these locations:";
-    for (std::string const& location : instances_after.failed_locations) {
+    for (std::string const& location : instances_after->failed_locations) {
       std::cout << " " << location << "\n";
```
Comment (devjgm) [inline google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc]: Same Q here about possibly `s/cout/cerr/`

File: google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc [hunk 3]
```diff
@@ -162,19 +174,22 @@ void CreateDevInstance(std::string project_id, int argc, char* argv[]) {
...
           [&instance_name](google::bigtable::admin::v2::Instance const& i) {
             return i.name() == instance_name;
           });
-  if (!instances.failed_locations.empty()) {
+  if (!instances->failed_locations.empty()) {
     std::cerr
         << "The service tells us it has no information about these locations:";
-    for (std::string const& location : instances.failed_locations) {
+    for (std::string const& location : instances->failed_locations) {
       std::cout << " " << location << "\n";
```
Comment (devjgm) [inline google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc]: here too. I realize that this isn't code that you changed in this PR, but it's close. Do we want to change it now? If we'd rather do this in a separate PR, that's fine with me, FWIW.

Comment (dopiera) [reply google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc]: How about I send you another PR with those issues addressed today then?

Comment (devjgm) [reply google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc]: SGTM.

Comment (dopiera) [reply google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc]: Can I have you LGTM on this PR then to submit it?

File: google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc [hunk 4]
```diff
@@ -324,41 +352,54 @@ void RunInstanceOperations(
...
+  for (auto const& instance : instances->instances) {
     std::cout << instance.name() << "\n";
   }
-  if (!instances.failed_locations.empty()) {
+  if (!instances->failed_locations.empty()) {
     std::cerr << "The Cloud Bigtable service reports that the following "
                  "locations are temporarily unavailable and no information "
                  "about instances in these locations can be obtained:\n";
-    for (auto& failed_location : instances.failed_locations) {
+    for (auto& failed_location : instances->failed_locations) {
```
[the remark this answers is not in the document]
Comment (devjgm) [reply google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc]: Did you want to leave this as `auto&` or change it in a followup PR also?

[the remark this answers is not in the document]
Comment (dopiera) [reply google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc]: Yup.
